File: src/main/java/sk/hudak/prco/service/WatchDogService.java
package sk.hudak.prco.service;

import sk.hudak.prco.api.EshopUuid;
import sk.hudak.prco.dto.WatchDogAddCustomDto;
import sk.hudak.prco.dto.WatchDogDto;
import sk.hudak.prco.dto.WatchDogNotifyUpdateDto;

import java.util.List;
import java.util.Map;

public interface WatchDogService {

    Long addNewProductToWatch(WatchDogAddCustomDto addDto);

    Map<EshopUuid,List<WatchDogDto>> findProductsForWatchDog();

    void notifyByEmail(List<WatchDogNotifyUpdateDto> toBeNotified);
}

File: src/main/java/sk/hudak/prco/service/WatchDogService.kt
package sk.hudak.prco.service

import sk.hudak.prco.api.EshopUuid
import sk.hudak.prco.dto.WatchDogAddCustomDto
import sk.hudak.prco.dto.WatchDogDto
import sk.hudak.prco.dto.WatchDogNotifyUpdateDto

interface WatchDogService {

    fun addNewProductToWatch(addDto: WatchDogAddCustomDto): Long?

    fun findProductsForWatchDog(): Map<EshopUuid, List<WatchDogDto>>

    fun notifyByEmail(toBeNotified: List<WatchDogNotifyUpdateDto>)
}
